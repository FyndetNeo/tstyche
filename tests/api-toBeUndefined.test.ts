import { expect, test } from "@jest/globals";
import { normalizeOutput } from "./__utils__/normalizeOutput.js";
import { spawnTyche } from "./__utils__/spawnTyche.js";

test("toBeUndefined", () => {
  const { status, stderr, stdout } = spawnTyche("api-toBeUndefined");

  expect(normalizeOutput(stdout)).toMatchSnapshot("stdout");
  expect(normalizeOutput(stderr)).toMatchSnapshot("stderr");

  expect(status).toBe(1);
});
