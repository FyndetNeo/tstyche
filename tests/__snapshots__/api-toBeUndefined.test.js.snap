// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toBeUndefined: stderr 1`] = `
"Error: The source type is 'string'.

   7 |   expect(returnsUndefined()).type.toBeUndefined();
   8 | 
>  9 |   expect(returnsString()).type.toBeUndefined();
     |                                ^
  10 | });
  11 | 
  12 | test("is NOT undefined?", () => {

       at ./__typetests__/toBeUndefined.tst.ts:9:32 ❭ is undefined?

Error: The source type is 'undefined'.

  13 |   expect(returnsString()).type.not.toBeUndefined();
  14 | 
> 15 |   expect(returnsUndefined()).type.not.toBeUndefined();
     |                                       ^
  16 | });
  17 | 

       at ./__typetests__/toBeUndefined.tst.ts:15:39 ❭ is NOT undefined?

"
`;

exports[`toBeUndefined: stdout 1`] = `
"uses TypeScript <<version>> with ./tsconfig.json

fail ./__typetests__/toBeUndefined.tst.ts
  × is undefined?
  × is NOT undefined?

Targets:    1 failed, 1 total
Test files: 1 failed, 1 total
Tests:      2 failed, 2 total
Assertions: 2 failed, 2 passed, 4 total
Duration:   <<timestamp>>

Ran all test files.
"
`;
